rust/src/lib.rs

```diff
@@ -1834,7 +1790,94 @@ pub struct FfiScanProgress {
     denominator: u64,
 }
 
-/// Returns the scan progress derived from the current wallet state.
+/// A type representing the potentially-spendable value of unspent outputs in the wallet.
+///
+/// The balances reported using this data structure may overestimate the total spendable
+/// value of the wallet, in the case that the spend of a previously received shielded note
+/// has not yet been detected by the process of scanning the chain. The balances reported
+/// using this data structure can only be certain to be unspent in the case that
+/// [`Self::is_synced`] is true, and even in this circumstance it is possible that a newly
+/// created transaction could conflict with a not-yet-mined transaction in the mempool.
+///
+/// # Safety
+///
+/// - `account_balances` must be non-null and must be valid for reads for
+///   `account_balances_len * mem::size_of::<FfiAccountBalance>()` many bytes, and it must
+///   be properly aligned. This means in particular:
+///   - The entire memory range pointed to by `account_balances` must be contained within
+///     a single allocated object. Slices can never span across multiple allocated objects.
+///   - `account_balances` must be non-null and aligned even for zero-length slices.
+///   - `account_balances` must point to `len` consecutive properly initialized values of
+///     type [`FfiAccountBalance`].
+/// - The total size `account_balances_len * mem::size_of::<FfiAccountBalance>()` of the
+///   slice pointed to by `account_balances` must be no larger than `isize::MAX`. See the
+///   safety documentation of `pointer::offset`.
+/// - `scan_progress` must, if non-null, point to a struct having the layout of
+///   [`FfiScanProgress`].
+#[repr(C)]
+pub struct FfiWalletSummary {
+    account_balances: *mut FfiAccountBalance,
+    account_balances_len: usize,
+    chain_tip_height: i32,
+    fully_scanned_height: i32,
+    scan_progress: *mut FfiScanProgress,
+    next_sapling_subtree_index: u64,
```

Comment:
The protobuf constrains this to a `u32`, so we should do that here as well. I'll fix this in the next PR.

Reply:
Opened #118.